src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/NegateInstruction.cs

```diff
@@ -221,10 +187,8 @@ public static Instruction Create(Type type)
                 case TypeCode.Int16: return s_Int16 ?? (s_Int16 = new NegateCheckedInt16());
                 case TypeCode.Int32: return s_Int32 ?? (s_Int32 = new NegateCheckedInt32());
                 case TypeCode.Int64: return s_Int64 ?? (s_Int64 = new NegateCheckedInt64());
-                case TypeCode.Single: return s_Single ?? (s_Single = new NegateCheckedSingle());
-                case TypeCode.Double: return s_Double ?? (s_Double = new NegateCheckedDouble());
                 default:
-                    throw ContractUtils.Unreachable;
```

Comment:
Any bug that would have hit this assertion will now hit the same assertion in `NegateInstruction.Create()` so this hasn't been loosened.